runtime/cudaq/platform/default/rest/helpers/ionq/IonQServerHelper.cpp

```diff
@@ -123,6 +126,18 @@ IonQServerHelper::createJob(std::vector<KernelExecution> &circuitCodes) {
     // Construct the job message
     ServerMessage job;
     job["target"] = backendConfig.at("target");
+    // Add noise model config to the JSON job request
+    if (keyExists("noise_model")) {
+      if (backendConfig.at("target") == "simulator") {
+        nlohmann::json noiseModel;
+        noiseModel["model"] = backendConfig.at("noise_model");
+        job["noise"] = noiseModel;
+      } else {
+        printf("IonQ noise model setting is only valid for the 'simulator' "
+               "ionq-machine target. This will be ignored.\n");
```

Comment:
We should silently ignore this argument. Maybe use `cudaq::info` to output this message when we run a program  with the appropriate log level. The reasoning here is that this option will be given to the compiler, but the message will only be shown when running a compiled program.

So a user would compile using `nvq++ ... --ionq-noise-model <model>`, get not warnings/errors, but then see this massage when running the produced output `./a.out`. We need to make the driver/compiler validate these things

Reply:
Thanks @boschmitt. That's a good point. I've updated the PR to perform this validation at compile time as suggested.